Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/main/home/HomeFragmentViewModel.kt

```diff
@@ -91,28 +91,27 @@ class HomeFragmentViewModel @AssistedInject constructor(
         .map { it.toHeaderState() }
         .asLiveData(dispatcherProvider.Default)
 
-    val popupEvents: SingleLiveEvent<HomeFragmentEvents> by lazy {
-        SingleLiveEvent<HomeFragmentEvents>().apply {
-            if (!LocalData.isInteroperabilityShownAtLeastOnce) {
-                postValue(ShowInteropDeltaOnboarding)
-            } else {
-                launch {
-                    if (!LocalData.tracingExplanationDialogWasShown()) {
-                        postValue(
-                            ShowTracingExplanation(
-                                TimeVariables.getActiveTracingDaysInRetentionPeriod()
-                            )
+    val popupEvents = SingleLiveEvent<HomeFragmentEvents>()
+
+    fun showPopUpsOrNavigate() {
+        if (!LocalData.isInteroperabilityShownAtLeastOnce) {
+            popupEvents.postValue(ShowInteropDeltaOnboarding)
+        } else if (cwaSettings.lastChangelogVersion.value < BuildConfigWrap.VERSION_CODE) {
+            popupEvents.postValue(HomeFragmentEvents.ShowNewReleaseFragment)
+        } else {
+            launch {
```

Comment:
why 2 different launch blocks? (not needed)
also using "when" instead of if else blocks is good practice :older_woman:

Reply:
This is not code that I have written. I only moved 

`if (cwaSettings.lastChangelogVersion.value < BuildConfigWrap.VERSION_CODE) {
                postValue(HomeFragmentEvents.ShowNewReleaseFragment)
}`

upwards into the `else if` condition. I am also not sure why there are 2 `launch{}` blocks but I don't want to remove one because they might be there for a reason we don't know.